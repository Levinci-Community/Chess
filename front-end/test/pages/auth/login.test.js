import React from "react";
import { render } from "@testing-library/react";
import LoginPage from "../../../src/pages/auth/login";
import { BrowserRouter as Router } from "react-router-dom";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({
    t: (key) => key,
  }),
}));

test("renders LoginPage snapshot", () => {
    const { container } = render(
      <Router>
        <LoginPage />
      </Router>
    );
    expect(container).toMatchSnapshot();
  });
