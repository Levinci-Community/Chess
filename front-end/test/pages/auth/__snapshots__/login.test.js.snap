// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LoginPage snapshot 1`] = `
<div>
  <div
    class="chakra-container css-25ehpw"
  >
    <div
      class="chakra-stack css-1821gv5"
    >
      <div
        class="chakra-stack css-8g8ihq"
      >
        <div
          class="chakra-stack css-tl3ftk"
        >
          <div
            class="chakra-stack css-1igwmid"
          >
            <img
              alt="common.ute_chess_club"
              class="chakra-image css-1570kgy"
              src="/logo.png"
            />
            <p
              class="chakra-text css-1hf07bm"
            >
              common.ute_chess_club
            </p>
          </div>
        </div>
      </div>
      <div
        class="css-1b0byzm"
      >
        <div
          class="chakra-stack css-mbssy4"
        >
          <div
            class="chakra-stack css-5bh0zh"
          >
            <h2
              class="chakra-heading css-0"
            >
              auth.login_to_your_account
            </h2>
          </div>
          <div
            class="chakra-stack css-1gnc3qn"
          >
            <div
              class="chakra-form-control css-0"
              role="group"
            >
              <label
                class="chakra-form__label css-uvho5g"
                for="username"
                id="field-:r0:-label"
              >
                auth.username
              </label>
              <input
                aria-required="true"
                class="chakra-input css-0"
                id="username"
                required=""
                type="username"
              />
            </div>
            <div
              class="chakra-form-control css-0"
              role="group"
            >
              <label
                class="chakra-form__label css-uvho5g"
                for="password"
                id="field-:r2:-label"
              >
                auth.password
              </label>
              <div
                class="chakra-input__group css-bx0blc"
                data-group="true"
              >
                <div
                  class="chakra-input__right-element css-dp9r5f"
                >
                  <button
                    aria-label="auth.reveal_password"
                    class="chakra-button css-p7mkev"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 20 20"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M10 12a2 2 0 100-4 2 2 0 000 4z"
                      />
                      <path
                        clip-rule="evenodd"
                        d="M.458 10C1.732 5.943 5.522 3 10 3s8.268 2.943 9.542 7c-1.274 4.057-5.064 7-9.542 7S1.732 14.057.458 10zM14 10a4 4 0 11-8 0 4 4 0 018 0z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </button>
                </div>
                <input
                  aria-required="true"
                  autocomplete="current-password"
                  class="chakra-input css-0"
                  id="password"
                  name="password"
                  required=""
                  type="password"
                />
              </div>
            </div>
          </div>
          <div
            class="chakra-stack css-1y3f6ad"
          >
            <div
              class="css-17xejub"
            />
            <button
              class="chakra-button css-4xx2wk"
              type="button"
            >
              auth.forgot_password
            </button>
          </div>
          <div
            class="chakra-stack css-1kkt86i"
          >
            <button
              class="chakra-button css-4xx2wk"
              type="button"
            >
              auth.login
            </button>
            <div
              class="chakra-stack css-1igwmid"
            >
              <hr
                aria-orientation="horizontal"
                class="chakra-divider css-1upb9tn"
              />
              <p
                class="chakra-text css-a2wsy3"
              >
                auth.or
              </p>
              <hr
                aria-orientation="horizontal"
                class="chakra-divider css-1upb9tn"
              />
            </div>
            <button
              class="chakra-button css-4xx2wk"
              type="button"
            >
              <p
                class="chakra-text css-1q07luh"
              >
                Google
              </p>
              <svg
                class="chakra-icon css-onkibi"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <g
                  transform="matrix(1, 0, 0, 1, 27.009001, -39.238998)"
                >
                  <path
                    d="M -3.264 51.509 C -3.264 50.719 -3.334 49.969 -3.454 49.239 L -14.754 49.239 L -14.754 53.749 L -8.284 53.749 C -8.574 55.229 -9.424 56.479 -10.684 57.329 L -10.684 60.329 L -6.824 60.329 C -4.564 58.239 -3.264 55.159 -3.264 51.509 Z"
                    fill="#4285F4"
                  />
                  <path
                    d="M -14.754 63.239 C -11.514 63.239 -8.804 62.159 -6.824 60.329 L -10.684 57.329 C -11.764 58.049 -13.134 58.489 -14.754 58.489 C -17.884 58.489 -20.534 56.379 -21.484 53.529 L -25.464 53.529 L -25.464 56.619 C -23.494 60.539 -19.444 63.239 -14.754 63.239 Z"
                    fill="#34A853"
                  />
                  <path
                    d="M -21.484 53.529 C -21.734 52.809 -21.864 52.039 -21.864 51.239 C -21.864 50.439 -21.724 49.669 -21.484 48.949 L -21.484 45.859 L -25.464 45.859 C -26.284 47.479 -26.754 49.299 -26.754 51.239 C -26.754 53.179 -26.284 54.999 -25.464 56.619 L -21.484 53.529 Z"
                    fill="#FBBC05"
                  />
                  <path
                    d="M -14.754 43.989 C -12.984 43.989 -11.404 44.599 -10.154 45.789 L -6.734 42.369 C -8.804 40.429 -11.514 39.239 -14.754 39.239 C -19.444 39.239 -23.494 41.939 -25.464 45.859 L -21.484 48.949 C -20.534 46.099 -17.884 43.989 -14.754 43.989 Z"
                    fill="#EA4335"
                  />
                </g>
              </svg>
            </button>
            <p
              class="chakra-text css-bmfe2s"
            >
              auth.dont_have_account
              <a
                class="chakra-link css-1o61trh"
                href="/register"
              >
                auth.register
              </a>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
